### ISSUE TITLE
Ambiguity in README regarding form of linear operator

### ISSUE THREAD

**Issue by samuelpowell** [JuliaLang/IterativeSolvers.jl#54]
README states that "A is not explicitly typed, but must either be a KrylovSubspace or support multiplication * or function composition (apply) that behave as necessary to produce the correct mapping on the vector space".

This led me to believe that I could do something such as the following:

```
J = rand(10,100); b = rand(100);
A(x) = J'*(J*x)
cg(A,rand(b))
```

But this fails as the package will eventually call ```eltype``` on the function ```A```.

I have no problems using the solvers with an instance of ```MatrixFcn```, but would like to understand if I have misunderstood the quoted text, or if it should be modified?

**Comment by jiahao** [comment]
It's true that the interface of `A` is not well defined and we should document it explicitly.

**Comment by jiahao** [comment]
Should be addressed by e609b03 for #63
